src/core/payments/index.ts

```diff
@@ -257,36 +257,29 @@ export async function depositUSDFC(
   synapse: Synapse,
   amount: bigint
 ): Promise<{
-  approvalTx?: string
   depositTx: string
 }> {
-  const paymentsAddress = synapse.getPaymentsAddress()
+  const needsAllowanceUpdate = (await checkAllowances(synapse)).needsUpdate
+  const amountMoreThanCurrentAllowance =
+    (await synapse.payments.allowance(synapse.getPaymentsAddress(), TOKENS.USDFC)) < amount
 
-  // Step 1: Check current allowance
-  const currentAllowance = await synapse.payments.allowance(paymentsAddress, TOKENS.USDFC)
+  let tx = {} as ethers.TransactionResponse
```

Comment:
nit, but this should help us ensure it's being set properly in the following code after branching and error cases/etc.

```suggestion
  let tx: ethers.TransactionResponse 
```